### ISSUE TITLE
Honor proxy settings during installation

### ISSUE THREAD

**Issue by bersbersbers** [pyppeteer/pyppeteer#307]
I just spent an hour trying to figure out why the install would stall forever after
> [W:pyppeteer.chromium_downloader] Starting Chromium download. Download may take a few minutes.

while `wget` and `curl` work without a problem.

The reason is that `urllib3` does not handle proxy information in environment variables, see https://github.com/urllib3/urllib3/issues/1785

I had to replace `urllib3.PoolManager()` by `urllib3.ProxyManager("http://my-proxy")` here
https://github.com/pyppeteer/pyppeteer/blob/6464ade4d9535927bae7f015b4fe35ff07807926/pyppeteer/chromium_downloader.py#L78
and then it worked fine - but it took me a while to sort this out.

**Comment by Mattwmaster58** [comment]
Interesting, it does indeed seem that urllib3 doesn't honor system proxy settings. This isn't intended behaviour I don't think but I'll be honest this is a low priority issue compared to the rest.

**Comment by gthieleb** [comment]
Seems this is intended by design urllib3/urllib3/issues/1785

**Comment by bersbersbers** [comment]
> [urllib3/urllib3/issues/1785](https://github.com/urllib3/urllib3/issues/1785)

@gthieleb yes, that is the issue I posted 18 months ago, and there hasn't been any activity since. Anything I might have missed there specifically?

**Comment by gthieleb** [comment]
@bersbersbers I dont think they will resolve this soon. Any chance your workaround packed in a small PR for pyppeteer? Some simple checking for the environment environment vars and selecting the correct manager.

**Comment by bersbersbers** [comment]
> I dont think they will resolve this soon

I agree.

> Any chance your workaround packed in a small PR for pyppeteer? Some simple checking for the environment environment vars and selecting the correct manager.

No, sorry - I have moved on from the project where I encountered that issue, and I hate to say that I learned from past contributions to other projects that there is no such thing as a small PR ;) setting up the environment, adopting the right coding style, find out which environment variables to check for, add tests, etc., that's not something I will have time for unfortunately. Hacking in the two lines of code reading and using one single environment variable shouldn't be too hard for anyone using pyppeteer.
